lib/plugins/paragon-webpack-plugin/ParagonWebpackPlugin.js

```diff
@@ -0,0 +1,230 @@
+    // parse file as html document
+    const document = parse5.parse(originalSource, {
+      sourceCodeLocationInfo: true,
+    });
+
+    // find the body element
+    const scriptInsertionPoint = this.findScriptInsertionPoint({
+      document,
+      originalSource,
+    });
+
+    // create Paragon script to inject into the HTML document
+    const paragonScript = `
+      <script type="text/javascript">
+        var PARAGON_THEME = ${JSON.stringify(scriptContents, null, 2)};
+      </script>
+    `;
+
+    // insert the Paragon script into the HTML document
+    const newSource = new sources.ReplaceSource(
+      new sources.RawSource(originalSource),
+      'index.html',
+    );
+    newSource.insert(scriptInsertionPoint, this.minifyScript(paragonScript));
+    return newSource;
+  }
+
+  apply(compiler) {
+    compiler.hooks.thisCompilation.tap('ParagonWebpackPlugin', (compilation) => {
+      compilation.hooks.processAssets.tap(
+        {
+          name: 'ParagonWebpackPlugin',
+          stage: Compilation.PROCESS_ASSETS_STAGE_ADDITIONS,
+          additionalAssets: true,
+        },
+        () => {
+          const file = compilation.getAsset('index.html');
+          if (!file) {
+            return;
+          }
+          const {
+            coreCssAsset: paragonCoreCssAsset,
+            themeVariantCssAssets: paragonThemeVariantCssAssets,
+          } = this.getParagonCssAssetsFromCompilation(compilation);
+
+          const {
+            coreCssAsset: brandCoreCssAsset,
+            themeVariantCssAssets: brandThemeVariantCssAssets,
+          } = this.getParagonCssAssetsFromCompilation(compilation, { isBrandOverride: true });
+
+          let scriptContents;
+          const createEmptyScriptContentsIfUndefined = () => {
+            if (!scriptContents) {
+              scriptContents = {};
+            }
+          };
+          const originalSource = file.source.source();
+
+          if (paragonCoreCssAsset && Object.keys(paragonThemeVariantCssAssets).length > 0) {
+            createEmptyScriptContentsIfUndefined();
+            scriptContents.paragon = {
+              version: this.paragon.version,
+              themeUrls: {
+                core: paragonCoreCssAsset,
+                variants: paragonThemeVariantCssAssets,
+              },
+            };
+          }
+
+          if (brandCoreCssAsset && Object.keys(brandThemeVariantCssAssets).length > 0) {
+            createEmptyScriptContentsIfUndefined();
+            scriptContents.brand = {
+              version: this.brand.version,
+              themeUrls: {
+                core: brandCoreCssAsset,
+                variants: brandThemeVariantCssAssets,
+              },
+            };
+          }
+
+          const newSource = this.insertParagonScriptIntoDocument({
+            originalSource,
+            coreCssAsset: paragonCoreCssAsset,
+            themeVariantCssAssets: paragonThemeVariantCssAssets,
+            scriptContents,
+          });
```

Comment:
I think `insertParagonScriptIntoDocument` only uses `originalSource` and `scriptContents` variables, so you can remove others.

Reply:
Ah, yup. Good catch! Left over from a prior iteration of the implementation. Will remove.